### PR TITLE
Add built-in function sign

### PR DESCRIPTION
**Pull request by mooleetzi**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [X] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #2102 

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Add built-in function sign

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/vectorize/sign/sign_test.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
+package sign
```
Comment (broccoliSpicy) [inline pkg/vectorize/sign/sign_test.go:1]: could you please add the license header

File: pkg/vectorize/sign/sign.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+func sign[T constraints.Integer | constraints.Float](xs []T, rs []int8) []int8 {
+	for i := range xs {
+		if xs[i] > 0 {
+			rs[i] = 1
+		} else if xs[i] == 0 {
+			rs[i] = 0
+		} else {
+			rs[i] = -1
+		}
+		if rs[i] > 1 || rs[i] < -1 {
```
Comment (fengttt) [inline pkg/vectorize/sign/sign.go:58]: You just assigned rs[i] to -1, 0, 1.    So this test at line 58 cannot possibly be true.   Please remove.

Comment (mooleetzi) [reply pkg/vectorize/sign/sign.go:58]: Oh，got it，I‘ll remove them.

**Review by NTH19 — APPROVED** [review]

**Comment by broccoliSpicy** [comment]
use cgo and use this bit hack:
```c
sign = (v > 0) - (v < 0); // -1, 0, or +1
```

matrixone strives for high performance 

reference: 
Donald E. Knuth, The Art of Computer Programming , Volume 4A, Combinatorial Algorithms, Part 1 , Addison-Wesley, 2011, Section 7.1.3.

https://www.youtube.com/watch?v=ZusiKXcz_ac

https://ocw.mit.edu/courses/6-172-performance-engineering-of-software-systems-fall-2018/resources/lecture-3-bit-hacks/

**Comment by mooleetzi** [comment]
> use cgo and use this bit hack:
> 
> ```c
> sign = (v > 0) - (v < 0); // -1, 0, or +1
> ```
> 
> matrixone strives for high performance
> 
> reference: Donald E. Knuth, The Art of Computer Programming , Volume 4A, Combinatorial Algorithms, Part 1 , Addison-Wesley, 2011, Section 7.1.3.
> 
> https://www.youtube.com/watch?v=ZusiKXcz_ac
> 
> https://ocw.mit.edu/courses/6-172-performance-engineering-of-software-systems-fall-2018/resources/lecture-3-bit-hacks/

OK, Thx. I will modify them

**Comment by mooleetzi** [comment]
> I am sorry, but we are going through a function framework refactoring. This change is caught in between and must be adapted to new framework.
Could you plz tell me where to catch the new framework，i  cannot tell them apart

**Pull request closed without merging**